runtime/src/bank/partitioned_epoch_rewards/distribution.rs

```diff
@@ -98,42 +139,82 @@ impl Bank {
         rewards.len().saturating_sub(initial_len)
     }
 
-    /// store stake rewards in partition
-    /// return the sum of all the stored rewards
+    fn build_updated_stake_reward(
+        stakes_cache_accounts: &im::HashMap<Pubkey, StakeAccount<Delegation>>,
+        partitioned_stake_reward: PartitionedStakeReward,
+    ) -> Result<StakeReward, DistributionError> {
+        let stake_account = stakes_cache_accounts
+            .get(&partitioned_stake_reward.stake_pubkey)
+            .ok_or(DistributionError::AccountNotFound)?
+            .clone();
+
+        let (mut account, stake_state): (AccountSharedData, StakeStateV2) = stake_account.into();
+        let StakeStateV2::Stake(meta, _stake, flags) = stake_state else {
```

Comment:
I'm wondering if it's worth checking that `_stake.delegation.stake + partitioned_stake_reward.stake_reward_info.lamports == partitioned_stake_reward.stake.delegation.stake`.

I can't come up with a situation where it *wouldn't* be the case, provided that the stake program doesn't change delegations, which makes an assert the safer option, but we could also log, skip that stake, and return an error. What do you think?